### ISSUE TITLE
Feature, unfeasible|development target: detecting screen composition

### ISSUE THREAD

**Issue by Uncle808** [kershner/screenBloom#39]
Don't know which topic I commented it on, but it's extremely odd when you look at a colored pillar in a room of different color and bloom averages this color prominently and turns the lighting over.

I realized when I spend some time looking for a while at the teaser video on the homepage
https://thumbs.gfycat.com/AffectionateDistinctBanteng-mobile.mp4

The close up of the leaves, when bloom turns reddish-brown, and the green leaves with the gecko's shadow let me connect my impression to my uneasyness: There should be positional weighting in relation to backgrounds or background colors.

Backgrounds are underlining the picture, but averaging the colors with centerpiece objects winning majority over the background bloom is actually distracting from the object in question by bleeding out it's colors into the real surroundings.
Object recognition like google's deepnet would be really awesome to detect the composition, but that's totally out of the question.

**Comment by kershner** [comment]
Hey there, fascinating discussion.  Thanks for bringing it up.

I've often thought that [the color averaging algorithm](https://github.com/kershner/screenBloom/blob/master/app/modules/img_proc.py#L17) could be improved quite a bit using actual computer science/theory.

I'm not quite clear on exactly what you're proposing, however.  Are you suggesting some way to differentiate between the back/foreground of an image and then perform separate calculations on the two sections?  That sounds closer to extracting the **dominant** color vs the average, which  I've experimented with numerous times.  The algorithms have always proven to be way too slow and unpredictable for ScreenBloom's purposes.

If I'm misunderstanding you, please let me know.  An "explain like I'm an idiot" would be very welcome!

**Comment by Uncle808** [comment]
Yeah, kind of dominant/reccessive color, but more than that. Taking those theories in, I also mean the general composition. The image itself is leading the eye around as it's broken up into regions of interest, affecting the perception of colors.

I'm an absolute laymen in these terms, have zero knowledge and may be an extreme case of shadow preferring basement dweller. I just noticed what perception differences hurt me the most when running bloom.
